app/controllers/TestController.scala

```diff
@@ -0,0 +1,135 @@
+package controllers
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, ExecutionException, Future}
+import scala.util.{Failure, Success, Try}
+import scala.util.control.NonFatal
+import akka.http.scaladsl.model.StatusCodes
+import play.api.mvc._
+import play.api.data.{Form, Mapping}
+import play.api.data.Forms._
+import play.api.i18n.I18nSupport
+import scorex.crypto.encode.{Base16, Base58}
+import models._
+import scorex.crypto.hash.Blake2b256
+import services.{TransactionService, WalletService}
+import storage.LSMStorage
+
+@Singleton
+class TestController @Inject()(implicit ec: ExecutionContext, ts: TransactionService, ws: WalletService, lsmStorage: LSMStorage, cc: ControllerComponents) extends AbstractController(cc) with I18nSupport {
+
+  var wallet: Option[Wallet] = None
```

Comment:
wallet should not be stored in controller, but rather be requested from corresponding service (and it should not be var)